extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -68,6 +77,9 @@ export class AmpViewerIntegration {
 
           viewer.setMessageDeliverer(messaging.sendRequest.bind(messaging),
             dev().assertString(this.unconfirmedViewerOrigin_));
+
+          listenOnce(
```

Comment:
Tests needed. Might be of help - we expose `window.eventListeners` for testing purposes in `realWin` and `fakeWin` fixtures. E.g. you should be able to do something like:
```
win.eventListeners.fire({type: 'unload'});
```